agent-lib/src/main/java/org/terracotta/angela/agent/AgentController.java

```diff
@@ -200,48 +197,43 @@ public String getTsaLicensePath(InstanceId instanceId, TerracottaServer terracot
   }
 
   public boolean installTms(InstanceId instanceId, String tmsHostname, Distribution distribution, License license,
-                            TmsServerSecurityConfig tmsServerSecurityConfig, String kitInstallationName,
-                            TerracottaCommandLineEnvironment tcEnv, Collection<String> hostNames) {
+                            String kitInstallationName,
+                            TerracottaCommandLineEnvironment tcEnv, String hostName, String kitInstallationPath) {
     TmsInstall tmsInstall = tmsInstalls.get(instanceId);
     if (tmsInstall != null) {
       logger.debug("Kit for " + tmsHostname + " already installed");
       tmsInstall.addTerracottaManagementServer();
       return true;
-    } else {
-      logger.debug("Attempting to install kit from cached install for " + tmsHostname);
-      RemoteKitManager kitManager = new RemoteKitManager(instanceId, distribution, kitInstallationName);
 
-      if (kitManager.isKitAvailable()) {
-        File kitDir = kitManager.installKit(license, hostNames);
-        File workingDir = kitManager.getWorkingDir().toFile();
+    } else {
 
-        File tmcProperties = new File(kitDir, "/tools/management/conf/tmc.properties");
-        if (tmsServerSecurityConfig != null) {
-          enableTmsSecurity(tmcProperties, tmsServerSecurityConfig);
-        }
-        tmsInstalls.put(instanceId, new TmsInstall(distribution, kitDir, workingDir, tcEnv));
-        return true;
-      } else {
+      Optional<Dirs> dirs = Dirs.discover(instanceId, hostName, distribution, license, kitInstallationName, kitInstallationPath);
+      if (!dirs.isPresent()) {
         return false;
       }
+
+      // Note: do not rewrite tmc.properties
+      // kitLocation might point to a shared kit, and tests can use this location concurrently.
+      // this is up to the test / gradle / maven task to correctly prepare the kit upfront in case a specific kit lcoation is provided.
+
+      tmsInstalls.put(instanceId, new TmsInstall(distribution, dirs.get().kitDir, dirs.get().workingDir, tcEnv));
```

Comment:
Is/was something rewriting `tmc.properties`? I'm really not a fan of all this kit customization.

Reply:
yes sadly... Because angela API allows to customize security "per test". But this only works if `angela.kitCopy` is set to true or if the kit is downloaded from elsewhere and "scoped" per test.

Reply:
Right but that should be done with system props... not by mutating the kit 🤮

Reply:
Completely agree! Which can be done now thanks to angela 3.8.1 which allows to override env variables per test

Reply:
I agree too, but this is a TMC problem : if you look at the start tmc command you can't pass a system property to use a custom tmc.properties. So the user has two choices : changing the start script to add a custom location or edit the tmc.properties which is located in the kit. I believe we used the second approach because it was what happens usually in customers setups.
I don't remember that there were system properties to replace the tmc.properties file, and I don't think it is documented either, but if it's there it would be easy to add support for that in Angela

Reply:
That's ok: all TC scripts allow to pass some env variables, such as `JAVA_OPTS`.
And with the previous PR I did, we can now pass any env vars as overrides per process that will be considered (see my email, angela 3.1.8 and #114

Reply:
yes you can pass JAVA_OPTS but here is the problem with the tmc-start script:

```
export APP_NAME=tmc
export RUN_ARGS="--spring.config.location=$TMC_HOME/conf/tmc.properties"

# this will only happen if using sag installer
if [ -r "${BIN_DIR}/setenv.sh" ] ; then
  . "${BIN_DIR}/setenv.sh"
fi

if [ ! -d "$JAVA_HOME" ]; then
   echo "ERROR: JAVA_HOME must point to Java installation."
   echo "    $JAVA_HOME"
   exit 2
fi

JAVA="$JAVA_HOME/bin/java"

exec "$JAVA" $JAVA_OPTS -jar $TMC_HOME/lib/tms.jar $RUN_ARGS
```

as you can see, the RUN_ARGS is set to a fixed tmc properties location, and is passed at the end of the command line, so it will override the tmc properties location if you pass it in the JAVA_OPTS

Regarding TMC system properties, where could I find some description?